### PR TITLE
Switched to proper field name - matterCategories.

### PR DESCRIPTION
**Pull request by ntgn81**
Updated selectiveToJS to call toJS on the data it finds if needed

@yneves, Not mergeable, just my first pass at handling client categories. Looks like the CategoriesInput is not accepting `selectedCategories` as props yet. Right?

### CODE REVIEW

**Comment by ntgn81** [comment]
I realized that that part is not done, there's an issue created for that. Can you take a look into this? Just continue the on this branch if what I did makes sense!

**Comment by yneves** [comment]
I'll check!

**Comment by yneves** [comment]
@ntgn81 check it out

File: src/components/CategoriesInput.js [hunk 1]
```diff
@@ -17,14 +17,27 @@ export default class CategoriesInput extends Component {
...
+    open: false
   };
 
+  constructor(props) {
+    super();
+    this.state.selectedCategories = props.selectedCategories || Immutable.List();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (!Immutable.is(nextProps.selectedCategories, this.props.selectedCategories)) {
```
Comment (ntgn81) [inline src/components/CategoriesInput.js]: This expects selectedCategories to be immutable, I did the change in selectiveToJS to convert the Immutable.List to array. We should probably revert that then?

Comment (ntgn81) [inline src/components/CategoriesInput.js]: We could probably be clearer in proptypes with something like Proptypes.instanceOf(Immutable.List)

Comment (yneves) [inline src/components/CategoriesInput.js]: What do you prefer? I used the immutable because of the easier comparison.

Comment (ntgn81) [inline src/components/CategoriesInput.js]: Immutable is good for me, as long as we can enforce it with proptypes

Comment (yneves) [inline src/components/CategoriesInput.js]: Sure, it's done!

**Comment by yneves** [comment]
I think this is good to go now. Matter's categories are being saved and displayed properly.

**Comment by ntgn81** [comment]
Great!

**Pull request merged**